### PR TITLE
fuzz: fix unused variable compiler warning in addrdb

### LINKED ISSUE #19549: Build warning in `test/fuzz/addrdb.cpp`: unused variable
Seen both locally and in the Travis fuzz build:
```cpp
test/fuzz/addrdb.cpp:20:21: warning: unused variable 'ban_entry' [-Wunused-variable]
    const CBanEntry ban_entry = [&] {
                    ^
1 warning generated.
```

### PR DESCRIPTION
**Pull request by jonatack**
Fixes #19549

### CODE REVIEW

File: src/test/fuzz/addrdb.cpp [hunk 1]
```diff
@@ -17,6 +17,8 @@ void test_one_input(const std::vector<uint8_t>& buffer)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
 
+    // `ban_entry` is no longer used, but the harness still needs to consume the
+    // same data that it did previously to not invalidate existing seeds.
```
Comment (MarcoFalke) [inline src/test/fuzz/addrdb.cpp:21]: Maybe comment here "The point of the code is to exercise all CBanEntry ctors :)" from https://github.com/bitcoin/bitcoin/issues/19549#issuecomment-660552734

File: src/test/fuzz/addrdb.cpp [hunk 2]
```diff
@@ -32,4 +34,5 @@ void test_one_input(const std::vector<uint8_t>& buffer)
         }
         return CBanEntry{};
     }();
+    (void)ban_entry;
```
Comment (MarcoFalke) [inline src/test/fuzz/addrdb.cpp:37]: and then comment this as 
```suggestion
    (void)ban_entry; // currently unused
```

**Review by maflcko — APPROVED** [review]
The rationale "not invalidate existing seeds" is a nice-to-have, not a strict rule. I think it is enough to say that the constructors should be exercised, but the entry itself is currently unused.

**Comment by MarcoFalke** [comment]
ACK either way

cc @practicalswift

**Comment by jonatack** [comment]
Thank @MarcoFalke, updated with your feedback.

**Comment by MarcoFalke** [comment]
review re-ACK 1cdc2a642b8b5dab7a830b5d7262add5363400db

**Review by hebasto — APPROVED** [review]
ACK 1cdc2a642b8b5dab7a830b5d7262add5363400db, tested on Linux Mint 20 (x86_64, clang 10.0).

**Comment by practicalswift** [comment]
ACK 1cdc2a642b8b5dab7a830b5d7262add5363400db

Thanks for fixing :)

**Pull request merged**